sdk/mgmtcommon/ClientRuntime/ClientRuntime/Utilities/HttpRequestSanitizer.cs

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+
+namespace Microsoft.Rest.Utilities
+{
+    /// <summary>
+    /// Sanitizer used internall by <see cref="HttpRequestMessageWrapper"/>.
+    /// </summary>
+    internal class HttpRequestSanitizer
+    {
+        public static readonly HttpRequestSanitizer Default = new HttpRequestSanitizer();
+
+        private readonly string _redactedPlaceholder = "REDACTED";
+        private readonly HashSet<string> _headersToSanitize = new HashSet<string>() { "authorization" };
```

Comment:
You should specify a comparer for this HashSet to make sure that all comparisons are done lowercase. You might be able to remove the `ToLowerCase()` call below.